packages/backend-defaults/src/entrypoints/cache/cacheServiceFactory.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CacheManager } from '@backstage/backend-common';
+import {
+  coreServices,
```

Comment:
I am more and more inclined to suggest renaming `coreServices` to `defaultServices`... Are there any conceptual differences between `core` and `default` services? It may also make sense to move the definitions to this package if we rename to defaultServices.

Reply:
Oh that's a bit of a bigger discussion I guess. We've called them core services all the time - the ones that are in `backend-plugin-api`. That's where the nomenclature comes from, irrespective of if `backend-defaults` also exists as a concept or not. They are the "core set of interfaces that you can consume". The default _implementations_ of those core interfaces live in `backend-defaults`, but you may bring your own.

Reply:
Is it right to say core services are the ones the backend application can't function without? Without a clear definition of what makes a service "core", I think it would simplify things if we considered them all as services with standard implementations since they're essential to a minimal backend. What about root services? Are they all core? Maybe it's just me, but do you remember other people asking about this or making confusion about core, root, and plugins services 🤔 ?

Reply:
> Is it right to say core services are the ones the backend application can't function without?

They are the ones that backend-plugin-api expose. No more, no less. They are owned by the core maintainers, and we guarantee that they have sensible definitions out of the box. It's "our sdk" so to speak. I don't think the definition is more complex than that. :)

Maybe it's more useful to think in terms of, what do we move to backend-plugin-api and what do we leave outside? We generally put things in there that we feel are so foundational, and useful for so many plugins, that we want to make it part of the central guaranteed offering. Events are such a thing that started outside as a plugin, but then got considered for "adoption" as a foundational piece, because it really is a foundation that others should be encouraged to use and share information between each other with.

Another thing that is special about many (but not all) core services is that they really don't have anywhere else to live. We don't have a logging plugin that can hold the logger.

> Without a clear definition of what makes a service "core", I think it would simplify things if we considered them all as services with standard implementations since they're essential to a minimal backend.

They're not though. It's just a toolbox. The things that we want to offer since we want to encourage people to build upon those primitives instead of reinventing wheels all over. Arguably not even config and logger are "essential", you just use them if you need them and we encourage you to try to leverage that instead of env vars and console logs, because it makes your plugin a better citizen.

> What about root services? Are they all core?

No, being root or plugin scoped is a separate orthogonal concern. The service itself chooses what makes sense in its own situation. That only deals with the "lifecycle" / instantiation of the service.

> Maybe it's just me, but do you remember other people asking about this or making confusion about core, root, and plugins services 🤔 ?

Hm not sure. The root/plugin thing I don't think I've heard. Figuring out where your deps live, yeah maybe? It's easy to find `coreServices`, people quickly pick up on that and enjoy the IDE completion helping them find what's available. But then, why isn't the catalog client in there for example? Sure, that gets a bit philosophical then, why is that not considered core but some others are? I kinda have an answer, but it's not always clear from the outside.